src/libsystemd/sd-id128/sd-id128.c

```diff
@@ -121,24 +121,56 @@ _public_ int sd_id128_string_equal(const char *s, sd_id128_t id) {
         return sd_id128_equal(parsed, id);
 }
 
-_public_ int sd_id128_get_machine(sd_id128_t *ret) {
+int id128_get_machine_at(int dir_fd, sd_id128_t *ret) {
```

Comment:
This function is where we should use `chase()` (or a helper) to to symlink chasing. I would also rename `dir_fd` to `rfd` here to indicate that we see it as a root directory file descriptor.